app/main.py

```diff
@@ -1,25 +1,80 @@
+from __future__ import annotations
 from abc import ABC
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: Visitor, name: str) -> None:
```

Comment:
Do not limit the use of `owner`, `instance` arguments, and return type for `__get__` to only instances of the `Visitor` class. Replace type annotation where you use `Visitor` to `Any` in the `IntegerRange` class, it may be useful to reuse this class.